src/main/common/sdft.c

```diff
@@ -130,7 +130,10 @@ FAST_CODE void sdftWinSq(const sdft_t *sdft, float *output)
     float re;
     float im;
 
-    for (int i = (sdft->startBin + 1); i < sdft->endBin; i++) {
+    const int start = MAX(sdft->startBin, 1);
+    const int end = MIN(sdft->endBin, SDFT_BIN_COUNT - 2);    
+
+    for (int i = start; i <= end; i++) {
         val = sdft->data[i] - 0.5f * (sdft->data[i - 1] + sdft->data[i + 1]); // multiply by 2 to save one multiplication
```

Comment:
sdft->data[i+1] will go out of bounds of the array if end = SDFT_BIN_COUNT - 1.

Reply:
That's why `end` is always smaller than `SDFT_BIN_COUNT - 1`

Reply:
![image](https://user-images.githubusercontent.com/46289813/154105616-e5d635ca-cff7-4467-8fb4-0120624053c1.png)

Reply:
Look closely

Reply:
Lets look at a min hz of 60 and max hz of 400. dynNotch.max = 400, in this case sdftResolutionHz = 11.11111111, 400/11.1111111 = 36, 36 + 0.5 = 36.5, now comparing SDFT_BIN_COUNT - 1 to 36 we get 36 -1 compared to 36.5, so sdftEndBin gets set to 35.
The array is of size 36, so the maximum bin is bin 35. So we run a for loop, looping through the arrays where 35 is the max value of the for loop, thats fine, until you do a sdft->data[i+1]. That begins to look at bin number 36 which doesn't exist.

Reply:
Like I said multiple times, check line 134. `const int end` is `34` at max, so nothing will go out of bounds.